### PR TITLE
feat: integrate schema classification

### PR DESCRIPTION
**Pull request by vjerci**
## Description
this pr refractores detections so that they live inside `/pkg/report/detections`
It also adds support for generic `DatatTypeable` interface handling, allowing helper functions to support different datatype structures.
It integrates datatypes into classification flow

## Checklist

- [ ] I've added test coverage that shows my fix or feature works as expected.
- [ ] I've updated or added documentation if required.
- [ ] I've included usage information in the description if CLI behavior was updated or added.
- [x] PR title follows [Conventional Commits](https://www.conventionalcommits.org/) format

### CODE REVIEW

File: pkg/report/schema/datatype/datatype.go
Comment (vjerci) [inline pkg/report/schema/datatype/datatype.go:18]: @elsapet I've changed a bit how datatypes work, I think this file is important for you.

File: pkg/classification/schema/schema_test.go
Comment (vjerci) [inline pkg/classification/schema/schema_test.go:45]: @elsapet This file has also been changed a bit

File: pkg/report/schema/datatype/datatype.go
Comment (cfabianski) [inline pkg/report/schema/datatype/datatype.go:21]: Could you add a `SimpleType` here too so that we can pass it to the classification engine?

File: pkg/report/schema/datatype/datatype.go
Comment (cfabianski) [inline pkg/report/schema/datatype/datatype.go:24]: Not sure I understand that.

**Review by cfabianski — COMMENTED** [review]
A lot of incidental changes. Left 2 comments but looks good in general.
@elsapet I'll let you approve if you are okay with that

File: pkg/report/schema/datatype/datatype.go
Comment (vjerci) [reply pkg/report/schema/datatype/datatype.go:24]: It is just something I've copied from previous workings.
it is something we needed for languages datatypes, If you mark it as helper it doesn't get exported.
As far as i recall it is used for taking care of duplicates, and I'm not sure if simply deleting it from parent properties was doable.

File: pkg/report/schema/datatype/datatype.go
Comment (vjerci) [reply pkg/report/schema/datatype/datatype.go:21]: Naming is a bit wierd here but in my opinion fixing it is outside of scope of this pr.
`Type` is actually simpletype while `TextType` is full type
So `type User struct` would have `type` of `object` and `TextType` of `User` .

Comment (cfabianski) [reply pkg/report/schema/datatype/datatype.go:21]: What is name then here?

**Review by elsapet — APPROVED** [review]

File: pkg/report/schema/datatype/datatype.go
Comment (vjerci) [reply pkg/report/schema/datatype/datatype.go:21]: i think name is also `user`

``` 
type User struct {
    id int
    address string
}
```

from this data we should have 
``` 
DataType{
	Name: "User",
	Type: schema.SimpleTypeObject,
	TextType: "User",
	Properties: map[string]DataTypable{
		"id": &DataType{
			Name: "id",
			Type: schema.SimpleTypeNumber,
			TextType: "int",
		},
		"address": &DataType{
			Name: "address",
			Type: schema.SimpleTypeString,
			TextType: "string",
		},
	},
} 
```

**Pull request merged**
